persist/fs/commitlog/writer.go

```diff
@@ -154,19 +157,38 @@ func (w *writer) isOpen() bool {
 	return w.chunkWriter.fd != nil
 }
 
+func (w *writer) logEntryIndex(seriesUniqueIndex uint64) uint64 {
+	// Special case the zeroth value as we compare the value
+	// at the entry of the seen cache with the index being put
+	// at the location - since all entries in the seen cache when
+	// not set are zero the value being used to compare whether
+	// this result has been seen or not cannot be zero itself
+	return seriesUniqueIndex + 1
+}
+
 func (w *writer) Write(
 	series Series,
 	datapoint ts.Datapoint,
 	unit xtime.Unit,
 	annotation ts.Annotation,
 ) error {
+	logEntryIndex := w.logEntryIndex(series.UniqueIndex)
```

Comment:
I was more concerned about the thrashing in the seenCache once go past capacity. Using the modulo would lead to a cascade of thrashing, any series could thrash an entry owned by another one. The hash approach would minimise that thrashing by being deterministic about which ids thrashed each other. Then again, it could also cause a lot more thrashing because two ids with the same hash could be written to very frequently. 

Might be best to just do the simple thing (like in the PR) and track a metric for thrashing, and address if it becomes an issue?